src/providers/object-provider.js

```diff
@@ -75,71 +74,122 @@ export default class YamcsObjectProvider {
 
     fetchTelemetryDictionary() {
         if(this.dictionaryPromise === undefined) {
-            this.dictionaryPromise = this.fetchMdbApi('containers').then(containers => {
-                return this.fetchMdbApi('parameters?details=yes&limit=1000&recurse=yes')
+            let url = this.getMdbUrl('space-systems')
+            this.dictionaryPromise = accumulateResults(url, 'spaceSystems', []).then(spaceSystems => {
+                return this.fetchMdbApi('parameters?details=yes&limit=1000')
                     .then(parameters => {
-                        let paramMap = {};
-                        if ('parameters' in parameters) {
+                        /* Sort the space systems by name, so that the
+                           children of the root object are in sorted order. */
+                        spaceSystems.sort((a, b) => {
+                            a.name.localeCompare(b.name)
+                        })
+                        spaceSystems.forEach(spaceSystem => {
+                            this.addSpaceSystem(spaceSystem)
+                        })
+                        if (parameters.parameters !== undefined) {
```

Comment:
Since we expect parameters.parameters to be an Array, could we use Array.isArray(parameters.parameters), instead of just checking if its undefined.

Reply:
That would be slightly slicker. Since Andrew's already approved, I will leave refinements up to the OpenMCT team, I think. In practice, because Yamcs has a fixed output format based on a Protobuf definition, the "parameters" attribute will either be missing or an array.